packages/ra-ui-materialui/src/input/InPlaceEditor/InPlaceEditor.stories.tsx

```diff
@@ -0,0 +1,372 @@
+    children,
+    dataProvider = fakeRestDataProvider(
+        { users: [{ id: 1, name: 'John Doe', age: 25, type: 'customer' }] },
+        process.env.NODE_ENV !== 'test',
+        500
+    ),
+}) => (
+    <TestMemoryRouter>
+        <QueryClientProvider client={new QueryClient()}>
+            <DataProviderContext.Provider value={dataProvider}>
+                <UndoableMutationsContextProvider>
+                    <I18nContextProvider value={i18nProvider}>
+                        <ThemeProvider theme={createTheme()}>
+                            <NotificationContextProvider>
+                                <ResourceContextProvider value="users">
+                                    <ShowBase id={1}>{children}</ShowBase>
+                                    <Notification />
+                                </ResourceContextProvider>
+                            </NotificationContextProvider>
+                        </ThemeProvider>
+                    </I18nContextProvider>
+                </UndoableMutationsContextProvider>
+            </DataProviderContext.Provider>
+        </QueryClientProvider>
+    </TestMemoryRouter>
+);
+
+export const Basic = ({
+    delay,
+    updateFails,
+    mutationMode,
+    notifyOnSuccess,
+    showButtons,
+}: {
+    delay?: number;
+    updateFails?: boolean;
+    mutationMode?: 'optimistic' | 'pessimistic' | 'undoable';
+    notifyOnSuccess?: boolean;
+    showButtons?: boolean;
+}) => {
+    const dataProvider = fakeRestDataProvider(
+        { users: [{ id: 1, name: 'John Doe', age: 25, type: 'customer' }] },
+        process.env.NODE_ENV !== 'test'
+    );
+    const sometimesFailsDataProvider = {
+        ...dataProvider,
+        update: async (resource, params) => {
+            return new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    if (updateFails) {
+                        reject(new Error('Server error'));
+                    } else {
+                        resolve(dataProvider.update(resource, params));
+                    }
+                }, delay);
+            }) as any;
+        },
+    };
+
+    return (
+        <Wrapper dataProvider={sometimesFailsDataProvider}>
+            <Typography color="secondary">Text above</Typography>
+            <InPlaceEditor
+                source="name"
+                mutationMode={mutationMode}
+                notifyOnSuccess={notifyOnSuccess}
+                showButtons={showButtons}
+            />
+            <Typography color="secondary">Text below</Typography>
+        </Wrapper>
+    );
+};
+
+Basic.args = {
+    delay: 500,
+    updateFails: false,
+    mutationMode: 'pessimistic',
+    notifyOnSuccess: false,
+    showButtons: false,
+};
+Basic.argTypes = {
+    delay: { control: { type: 'number' } },
+    updateFails: { control: { type: 'boolean' } },
+    mutationMode: {
+        control: { type: 'select' },
+        options: ['optimistic', 'pessimistic', 'undoable'],
```

Comment:
When testing `undoable` mode, the record does not seem to be refreshed and I can't see my changes being saved (I see the dataProvider call in the console though)

Reply:
Probably because you didn't enable `notifyOnSuccess` at the same time.